### PR TITLE
CB-7147 - [ASRG] Refactor termination resource collection

### PR DESCRIPTION
**Pull request by pdarvasi**
Contains the following 2 commits:

CB-6584 - [ASRG] Termination - refactor downscale code
CB-7147 - [ASRG] Refactor termination resource collection and E2E test

### CODE REVIEW

File: cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/AzureResourceConnector.java [hunk 1]
```diff
@@ -236,35 +210,35 @@ private void checkTemplateDeployment(List<CloudResourceStatus> result, AzureClie
...
-                throw new CloudConnectorException(String.format("Could not delete resource group: %s", resourceGroupName), e);
-            } else {
-                return check(ac, Collections.emptyList());
+                try {
+                    retryService.testWith2SecDelayMax5Times(() -> {
+                        if (!client.resourceGroupExists(resourceGroupName)) {
+                            throw new ActionFailedException("Resource group not exists");
+                        }
+                        return true;
+                    });
```
Comment (gergopapi2) [inline cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/AzureResourceConnector.java:228]: This part could go into AzureUtils.

File: cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/AzureResourceConnector.java [hunk 2]
```diff
@@ -45,32 +34,22 @@
...
 import com.sequenceiq.cloudbreak.cloud.notification.PersistenceNotifier;
+import com.sequenceiq.cloudbreak.cloud.template.AbstractResourceConnector;
 import com.sequenceiq.cloudbreak.service.Retry;
 import com.sequenceiq.cloudbreak.service.Retry.ActionFailedException;
 import com.sequenceiq.common.api.type.AdjustmentType;
 import com.sequenceiq.common.api.type.ResourceType;
 
 @Service
-public class AzureResourceConnector implements ResourceConnector<Map<String, Map<String, Object>>> {
+public class AzureResourceConnector extends AbstractResourceConnector {
```
Comment (gergopapi2) [inline cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/AzureResourceConnector.java:47]: StartDatabaseServer and StopDatabaseServer are identical to the base implementation (NotImplemented), can be deleted.

File: cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/AzureResourceConnector.java [hunk 3]
```diff
@@ -45,32 +34,22 @@
...
 
     public static final String RESOURCE_GROUP_NAME = "resourceGroupName";
```
Comment (gergopapi2) [inline cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/AzureResourceConnector.java:49]: Can be package private.

File: cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/AzureUtils.java [hunk 4]
```diff
@@ -401,4 +433,21 @@ public void deleteManagedDisks(AzureClient azureClient, Collection<String> manag
...
+            LOGGER.info("Delete managed disks with id-s: {}", managedDiskIds);
+
+            Observable<String> deletionObservable = azureClient.deleteManagedDiskAsync(managedDiskIds)
+                    .doOnError(throwable -> {
+                        LOGGER.error("Error happened during the deletion of the managed disks ", throwable);
+                        throw new CloudbreakServiceException("Can't delete all managed disks: ", throwable);
+                    })
+                    .doOnCompleted(() -> LOGGER.debug("Delete managed disks completed successfully"))
+                    .subscribeOn(Schedulers.io());
+            deletionObservable.subscribe(disk -> LOGGER.debug("Deleting {}", disk));
```
[the remark this answers is not in the document]
Comment (pdarvasi) [reply cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/AzureUtils.java:449]: From SupportsBatchDeletion interface: "@return a representation of the deferred computation of this call returning the resource ID of each successfully deleted resource"

File: cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/AzureVirtualMachineService.java [hunk 5]
```diff
@@ -76,8 +76,15 @@ public void refreshInstanceViews(Map<String, VirtualMachine> virtualMachines) {
...
-        return !virtualMachineNames.containsAll(privateInstanceIds);
+        Set<String> virtualMachineNames = virtualMachines
+                .stream()
+                .map(VirtualMachine::name)
+                .collect(Collectors.toSet());
+        boolean hasMissingVm = !virtualMachineNames.containsAll(privateInstanceIds);
+        if (hasMissingVm) {
+            LOGGER.info("Fetched VM id-s ({}) do not contain one of the following id-s: {}", virtualMachineNames, privateInstanceIds);
+        }
+        return !hasMissingVm;
```
[the remark this answers is not in the document]
Comment (pdarvasi) [reply cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/AzureVirtualMachineService.java:87]: Nice catch!

**Pull request merged**